src/api/java/appeng/api/storage/cells/CellState.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 AlgorithmX2
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package appeng.api.storage.cells;
+
+/**
+ * Indicates the state of a cell
+ * 
+ * MISSING: No cell at all
+ * EMPTY: A cell without anything stored
+ * USED: Stored something, but neither types nor totally full
+ * TYPES_FULL: Available types exhausted
+ * FULL: Full cell, technically could have free types
+ * 
+ * @author yueh
+ *
+ */
+public enum CellState {
+	MISSING, EMPTY, USED, TYPES_FULL, FULL,
```

Comment:
Naming of the enum is inconsistent (and always was).
Is it the state of a cell, or is it the state of a slot in a drive or chest?
Because what is the state "USED"? NOT_EMPTY?
And should the absence of a cell not be indicated using a null enumeration value?